src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,652 @@
+    /// From a User perspective, this is the token being bought.
+    /// token1 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token1;
+
+    ////////////////////////////////////////////////////
+    ///             Storage Variables
+    ////////////////////////////////////////////////////
+
+    /**
+     * @notice For one `token0` from a Trader, how many `token1` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate0` is the WETH/stETH price.
+     * From a Trader's perspective, this is the buy price.
+     * From the ARM's perspective, this is the sell price.
+     * Rate is to 36 decimals (1e36).
+     * To convert to a stETH/WETH price, use `PRICE_SCALE * PRICE_SCALE / traderate0`.
+     */
+    uint256 public traderate0;
+    /**
+     * @notice For one `token1` from a Trader, how many `token0` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate1` is the stETH/WETH price.
+     * From a Trader's perspective, this is the sell price.
+     * From a ARM's perspective, this is the buy price.
+     * Rate is to 36 decimals (1e36).
+     */
+    uint256 public traderate1;
+
+    /// @notice cumulative total of all withdrawal requests included the ones that have already been claimed
+    uint120 public withdrawsQueued;
+    /// @notice total of all the withdrawal requests that have been claimed
+    uint120 public withdrawsClaimed;
+    /// @notice index of the next withdrawal request starting at 0
+    uint16 public nextWithdrawalIndex;
+
+    struct WithdrawalRequest {
+        address withdrawer;
+        bool claimed;
+        // When the withdrawal can be claimed
+        uint40 claimTimestamp;
+        // Amount of assets to withdraw
+        uint120 assets;
+        // cumulative total of all withdrawal requests including this one.
+        // this request can be claimed when this queued amount is less than or equal to the queue's claimable amount.
+        uint120 queued;
+    }
+
+    /// @notice Mapping of withdrawal request indices to the user withdrawal request data
+    mapping(uint256 requestId => WithdrawalRequest) public withdrawalRequests;
+
+    /// @notice Performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent)
+    /// 10,000 = 100% performance fee
+    /// 2,000 = 20% performance fee
+    /// 500 = 5% performance fee
+    uint16 public fee;
+    /// @notice The available assets the the last time performance fees were collected and adjusted
+    /// for liquidity assets (WETH) deposited and redeemed.
+    /// This can be negative if there were asset gains and then all the liquidity providers redeemed.
+    int128 public lastAvailableAssets;
+    /// @notice The account that can collect the performance fee
+    address public feeCollector;
+
+    address public liquidityProviderController;
+
+    uint256[42] private _gap;
```

Comment:
🟠 _Gap goblin arrives, points at gap._

Should be 43 now. We need lint tooling for this, lol.

Reply:
changed